packages/nova-lib/lib/modules/redux.js

```diff
@@ -1,25 +1,78 @@
+import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
+
+export const STORE_RELOADED = 'STORE_RELOADED';
+
+// create store, and implement reload function
+export const configureStore = (reducers, initialState = {}, middlewares) => {
+  let getReducers;
+  if (typeof reducers === 'function') {
+    getReducers = reducers;
+    reducers = getReducers();
+  }
+  reducers = typeof reducers === 'object' ? combineReducers(reducers) : reducers;
+  middlewares = Array.isArray(middlewares) ? middlewares : [middlewares];
+
+  const store = createStore(
+    // reducers
+    reducers,
+    // initial state
+    initialState,
+    // middlewares
+    compose(
+      applyMiddleware(...middlewares),
+      typeof window !== 'undefined' && window.devToolsExtension ? window.devToolsExtension() : f => f,
+    ),
+  );
+
+  store.reload = function reload(options = {}) {
+    if (typeof options.reducers === 'function') {
+      getReducers = options.reducers;
+      options.reducers = undefined;
+    }
+    if (!options.reducers && getReducers) {
+      options.reducers = getReducers();
+    }
+    if (options.reducers) {
+      reducers = typeof options.reducers === 'object' ? combineReducers(options.reducers) : options.reducers;
+    }
+    this.replaceReducer(reducers);
```

Comment:
yes.
for example, on client side,
currently i initialise a empty store object at first in nova:lib
the main reason is i don't want init store in nova:routing.
init a empty store in nova:lib it is good for other packages to access it.
so if you add several reducers from other package.
you must replaceReducer and let the store update with the added reducers